qutip/animation.py

```diff
@@ -0,0 +1,579 @@
+"""
+Functions to animate results of quantum dynamics simulations,
+"""
+__all__ = ['anim_wigner_sphere', 'anim_hinton', 'anim_sphereplot',
+           'anim_matrix_histogram', 'anim_fock_distribution', 'anim_wigner',
+           'anim_spin_distribution', 'anim_qubism', 'anim_schmidt']
+
+from . import (plot_wigner_sphere, hinton, sphereplot, matrix_histogram,
+               plot_fock_distribution, plot_wigner, plot_spin_distribution,
+               plot_qubism, plot_schmidt)
+from .solver import Result
+
+try:
+    import matplotlib.pyplot as plt
+except Exception:
+    pass
+
+
+def _result_state(obj):
+    if isinstance(obj, Result):
+        return obj.states
```

Comment:
`results` sometime do not save the states and `result.states` is None.
I would check that case and raise an error telling the users to add `options={"store_states": True}` to the simulation.

Reply:
I added an assertion. ```result.states``` returns a list so it raises an error when the length of it is zero.